tee-apps/sealed-bid-auction/src/blockchain/ViemIntentObserver.ts

```diff
@@ -87,21 +82,39 @@ export class ViemIntentObserver {
             this.logger.debug(`Auction winner: ${serialize(winningPrice)}`);
 
             if (winningPrice !== null && winningPrice.amountOut >= orderData.minAmountOut) {
+
                 const result: AuctionResult = {
+                    type: "winning-bid",
                     settlementReceiverAddress: winningPrice.settlementReceiverAddress,
                     amountOut: winningPrice.amountOut,
                     orderId,
-                    orderData
+                    signature: await this.signAuctionResult(
+                        BigInt(orderId), winningPrice.settlementReceiverAddress as `0x${string}`, winningPrice.amountOut
+                    )
                 }
 
-                await this.apiServer.notifySolvers(result, this.client.chain.id);
-                const txHash = await this.auctionCommiter.commitWinnerBid(result);
+                await this.apiServer.notifySolvers(result, this.blockchainClient.publicClient.chain.id);
 
-                this.logger.info(`I finished auction for orderId=[${orderId}] , notified solvers amd sent winningBid using tx=[${txHash}]`);
+                this.logger.info(`I finished auction for orderId=[${orderId}] and notified solvers`);
                 break;
             }
             
             await new Promise((resolve) => setTimeout(resolve, this.auctionPollingInterval));
         }
     }
+
+    private async signAuctionResult(orderId: bigint, winningSolver: `0x${string}`, amountOut: bigint) : Promise<string> {
+        const encoded = encodeAbiParameters(
```

Comment:
Ok so my bad, I realise I didn't explicitly mention in the design that the value should also be encoded according to EIP 712 before being signed (sorry for being a smart contract nerd that assumes it's obvious).
I can take care of this if you want.